test/powershell/Modules/Microsoft.PowerShell.Utility/Get-Random.Tests.ps1

```diff
@@ -2,48 +2,48 @@
-        @{ Name = 'both are int64.MaxValue plus a 32-bit integer'; Maximum = ([int64](([int]::MaxValue)+15)); Minimum = ([int64](([int]::MaxValue)+10)); GreaterThan = ([int64](([int]::MaxValue)+9)); LessThan = ([int64](([int]::MaxValue)+15)); Type = 'System.Int64' }
-        @{ Name = 'both are greater than int32.MaxValue without specified type, and max with quote'; Maximum = '100099000001'; Minimum = 100000000001; GreaterThan = ([int64]10000000000); LessThan = ([int64]100099000001); Type = 'System.Int64' }
-        @{ Name = 'both are greater than int32.MaxValue without specified type, and min with quote'; Maximum = 100000002230; Minimum = '100000002222'; GreaterThan = ([int64]100000002221); LessThan = ([int64]100000002230); Type = 'System.Int64' }
-        @{ Name = 'max is greater than int32.MaxValue without specified type'; Maximum = 90000000000; Minimum = 4; GreaterThan = ([int64]3); LessThan = ([int64]90000000000); Type = 'System.Int64' }
-        @{ Name = 'max is a double-precision number'; Maximum = 100.0; Minimum = $null; GreaterThan = -1.0; LessThan = 100.0; Type = 'System.Double' }
-        @{ Name = 'both are double-precision numbers, min is negative.'; Maximum = 0.0; Minimum = -100.0; GreaterThan = -101.0; LessThan = 0.0; Type = 'System.Double' }
-        @{ Name = 'both are double-precision number, max is positive, min is negative.'; Maximum = 100.0; Minimum = -100.0; GreaterThan = -101.0; LessThan = 100.0; Type = 'System.Double' }
-        @{ Name = 'max is a double-precision number, min is int32'; Maximum = 8.0; Minimum = 5; GreaterThan = 4.0; LessThan = 8.0; Type = 'System.Double' }
-        @{ Name = 'min is a double-precision number, max is int32'; Maximum = 8; Minimum = 5.0; GreaterThan = 4.0; LessThan = 8.0; Type = 'System.Double' }
-        @{ Name = 'max set to a special double number'; Maximum = 20.; Minimum = 0.0; GreaterThan = -1.0; LessThan = 20.0; Type = 'System.Double' }
-        @{ Name = 'max is double with quote'; Maximum = '20.'; Minimum = 0.0; GreaterThan = -1.0; LessThan = 20.0; Type = 'System.Double' }
-        @{ Name = 'max is double with plus sign'; Maximum = +100.0; Minimum = 0; GreaterThan = -1.0; LessThan = 100.0; Type = 'System.Double' }
-        @{ Name = 'max is double with plus sign and enclosed in quote'; Maximum = '+100.0'; Minimum = 0; GreaterThan = -1.0; LessThan = 100.0; Type = 'System.Double' }
-        @{ Name = 'both set to the special numbers as 1.0e+xx '; Maximum = $null; Minimum = 1.0e+100; GreaterThan = 1.0e+99; LessThan = ([double]::MaxValue); Type = 'System.Double' }
-        @{ Name = 'max is Double.MaxValue, min is Double.MinValue'; Maximum = ([double]::MaxValue); Minimum = ([double]::MinValue); GreaterThan = ([double]::MinValue); LessThan = ([double]::MaxValue); Type = 'System.Double' }
+        @{ Name = 'no params'; Maximum = $null; Minimum = $null; GreaterThan = -1; LessOrEqual = ([int32]::MaxValue); Type = 'System.Int32' }
```

Comment:
Why was `LessThan` changed to `LessOrEqual`?

Reply:
The debug.assert statements in the code a exlusive for lower bounds and inclusive for higher bounds thus we have changed LessThan into LessOrEqual to make it consistent with the current implementation.

Reply:
Yes, our tests do not match the code. One of the two must be corrected.

Reply:
Wait sorry i mixed things up. The assert statements are exclusive for higher bounds and inclusive for lower bounds. In any case I think I should just revert the changes made to the tests for the following reason : 
- The debug.assert statements should be moved into the actual random number generators but for that it also means looking at what numbers can be generated (inclusive bounds VS exclusive bounds)
I think that discussion can be had in another PR and this one can simply remain very pointed in scope :)
I have no issue opening the extra PR myself btw.

Reply:
@pougetat Let's wait @adityapatwardhan conclusion.

Reply:
@pougetat Sorry I reverted the commit when I wrote my previous comment. Having said that I still have it on a local branch so I can put it back easily depending on his conclusion. I should have simply reverted with an actual commit instead of dropping the commit altogether. In any case yes I agree we can wait for his conclusion :).

Reply:
I agree with having a separate PR for that. I just signed off.

Reply:
@pougetat for future reference, I do believe the random generators are inclusive on the lower bound, but exclusive on the higher bound. Don't recall the exact reason for it, but it's a pretty common pattern.

Reply:
It is documented https://docs.microsoft.com/en-us/dotnet/api/system.random.next?redirectedfrom=MSDN&view=netcore-2.2#System_Random_Next_System_Int32_System_Int32_